app/controllers/chapters_controller.rb

```diff
@@ -91,10 +91,10 @@ def new
   # GET /work/:work_id/chapters/1/edit
   def edit
     if params["remove"] == "me"
-      @chapter.pseuds = @chapter.pseuds - current_user.pseuds
+      @chapter.authors_to_remove = current_user.pseuds
```

Comment:
When you have a work with multiple creators and a chapter that _isn't_ by all the people on the work, under the chapter title will be a second byline saying "Chapter by creator." This part of the chapter is cached, and the cache wasn't expiring when a creator removed themselves from a chapter. This was happening because nothing _in_ the chapter itself was changing -- the creators for a chapter aren't in Chapter, but instead in Creatorship -- so the cache key, which is based on the chapter's updated_at date, wasn't expiring.

I tried adding a callback in the Chapter model to touch the chapter when pseuds were removed, but it was a damn lot of code. I tried touching the chapter using the before_destroy callback in the Creatorship model (here: https://github.com/otwcode/otwarchive/blob/master/app/models/creatorship.rb#L5), but surprise! That doesn't actually get called.

And that is how I ended up with this method -- setting authors_to_remove and then touching the chapter in the creation_observer.